eng/common/pipelines/ai-evals-tests.yml

```diff
@@ -0,0 +1,94 @@
+trigger: none
+
+pr:
+  branches:
+    include:
+      - main
+  paths:
+    include:
+      - .github/copilot-instructions.md
```

Comment:
What is the plan for running this in other repos? Should we start by having this only in the tools repo and run whenever the common instructions change? Or maybe only have it in the specs repo if the repo level instructions are changing more heavily there.